js/graphql.md

```diff
@@ -0,0 +1,1374 @@
+---
+---
+# The GraphQL Transform
+
+The GraphQL Transform is a library that simplifies the process of designing GraphQL APIs.
+In short, it allows a developer to design application backends with a simple `schema.graphql`
+which is then *transformed* into a AWS CloudFormation that implements the data model
+defined in `schema.graphql`. For example you might create the backend for a blog like this:
+
+```
+type Blog @model {
+  id: ID!
+  name: String!
+  posts: [Post] @connection(name: "BlogPosts")
+}
+type Post @model {
+  id: ID!
+  title: String!
+  blog: Blog @connection(name: "BlogPosts")
+  comments: [Comment] @connection(name: "PostComments")
+}
+type Comment @model {
+  id: ID!
+  content: String
+  post: Post @connection(name: "PostComments")
+}
+```
+
+When used along with tools like the Amplify CLI, the GraphQL Transform simplifies the process of 
+developing, deploying, and maintaining GraphQL APIs. With it, you define your API using the 
+GraphQL Schema Definition Language (SDL) and can then use automation to transform it into a fully 
+descriptive cloudformation template that implements the spec. The transform also provides a framework
+through which you can define you own transformers as `@directives` for custom workflows.
+
+## Quick Start
+
+Navigate into the root of a JavaScript, iOS, or Android project and run:
+
+```bash
+amplify init
+```
+
+Follow the wizard to create a new app. After finishing the wizard run:
+
+```bash
+amplify add api
+
+# Select the graphql option and when asked if you 
+# have a schema, say No.
+# Select one of the default samples. You can change it later.
+# Choose to edit the schema and it will open your schema.graphql in your editor.
+```
+
+You can leave the sample as is or try this schema.
+
+```
+type Blog @model {
+  id: ID!
+  name: String!
+  posts: [Post] @connection(name: "BlogPosts")
+}
+type Post @model {
+  id: ID!
+  title: String!
+  blog: Blog @connection(name: "BlogPosts")
+  comments: [Comment] @connection(name: "PostComments")
+}
+type Comment @model {
+  id: ID!
+  content: String
+  post: Post @connection(name: "PostComments")
+}
+```
+
+Once you are happy with your schema, save the file and click enter in your
+terminal window. If now error messages are thrown then you are good to go
+and can deploy your new API.
+
+```bash
+amplify push
+```
+
+Go to AWS CloudFormation to view it. You can also find your project assets in the amplify/backend folder under your API.
+
+Once the API is finsihed deploying, try going to the AWS AppSync console and
+running some of these queries in your new API's query page.
```

Comment:
@kaustavghosh06 it would be really nice to have AppSync available to the `amplify console` command here so that the customer could just run that and copy in these queries in one command.